src/main/java/com/faforever/api/forum/NodebbService.java

```diff
@@ -0,0 +1,102 @@
+package com.faforever.api.forum;
+
+import com.faforever.api.config.FafApiProperties;
+import com.faforever.api.user.UserDataSyncService;
+import com.faforever.api.user.UserUpdatedEvent;
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import lombok.RequiredArgsConstructor;
+import lombok.Value;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+import org.springframework.web.util.UriComponentsBuilder;
+
+import java.net.URI;
+import java.util.Optional;
+
+@Service
+@ConditionalOnProperty(value = "faf-api.nodebb.master-token")
+@Slf4j
+@RequiredArgsConstructor
+public class NodebbService implements UserDataSyncService {
+  private final RestTemplate restTemplate;
+  private final FafApiProperties properties;
+
+  @Override
+  public void userDataChanged(UserUpdatedEvent event) {
+    try {
+      getNodebbUserId(event.getId())
+        .ifPresentOrElse(
+          userId -> updateUserData(userId, event),
+          () -> log.info("User data not updated in NodeBB (User not found): {}", event)
```

Comment:
warn? Becuase shold not happen

Reply:
You only have a user account in NobdeBB after you first logged in via SSO. So 99% of our userbase will not have a user.